utils/staking-miner/src/monitor.rs

```diff
@@ -54,17 +63,63 @@ async fn ensure_no_previous_solution<
 	T: EPM::Config + frame_system::Config<AccountId = AccountId>,
 	B: BlockT,
 >(
-	ext: &mut Ext,
+	client: &WsClient,
+	at: B::Hash,
 	us: &AccountId,
 ) -> Result<(), Error<T>> {
-	use EPM::signed::SignedSubmissions;
-	ext.execute_with(|| {
-		if <SignedSubmissions<T>>::get().iter().any(|ss| &ss.who == us) {
-			Err(Error::AlreadySubmitted)
-		} else {
-			Ok(())
+	let indices_key = storage_value(EPM_MODULE_PREFIX, b"SignedSubmissionIndices");
+
+	let indices: SubmissionIndicesOf<T> = get_storage(client, rpc_params! {indices_key, at})
+		.await
+		.map_err::<Error<T>, _>(Into::into)?
+		.unwrap_or_default();
+
+	for (_score, id) in indices {
+		let key = storage_value_by_key::<Twox64Concat>(
+			EPM_MODULE_PREFIX,
+			b"SignedSubmissionsMap",
+			&id.encode(),
+		);
+
+		if let Some(submission) =
```

Comment:
I would wait until your other PR is merged (which you can do ASAP since it is all approved now) to rewrite this stuff with the new PRC helpers? I think the code would be cleaner.